### PR TITLE
Handle Request batchId

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [na] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

To be used with https://github.com/opentripplanner/otp-react-redux/pull/1252.

This PR corrects the persistence of trip requests (for applicable users) by persisting the `batchId` parameter provided by the UI client instead of creating a `batchId` that is not related to the search performed in the UI.

### CODE REVIEW

**Review by br648 — APPROVED** [review]
LGTM.

File: src/main/java/org/opentripplanner/middleware/controllers/api/OtpRequestProcessor.java [hunk 1]
```diff
@@ -190,12 +190,8 @@ private String proxyPost(Request request, Response response) {
...
                 * Other requests will still be proxied, just not stored.
                 */
-                QueryVariables queryVariables = getPOJOFromJSON(requestBody, Query.class).variables;
-
-                // Follows the method used in otp-ui core-utils storage.js
-                String randomBatchId = Integer.toString((int) (Math.random() * 1_000_000_000), 36);
-
-                if (!handlePlanTripResponse(randomBatchId, queryVariables, otpDispatcherResponse, otpUser)) {
+                Query query = getPOJOFromJSON(requestBody, Query.class);
+                if (!handlePlanTripResponse(query.batchId, query.variables, otpDispatcherResponse, otpUser)) {
```
Comment (miles-grant-ibigroup) [inline src/main/java/org/opentripplanner/middleware/controllers/api/OtpRequestProcessor.java:194]: I'm assuming `handlePlanTripResponse` is able to handle batch id missing?

**Review by miles-grant-ibigroup — APPROVED** [review]
Just making sure a missing `batchId` doesn't cause a crash but otherwise this looks nice and clean!

File: src/main/java/org/opentripplanner/middleware/controllers/api/OtpRequestProcessor.java (hunk 1 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/controllers/api/OtpRequestProcessor.java:194]: That is the case. With this PR,`batchId` is an optional field when parsing already-stored trip requests, and `handlePlanTripResponse` already sets `batchId` to "Batch Id not provided by caller." if not provided.

**Pull request merged**